x-pack/plugins/security_solution/server/endpoint/routes/resolver/docs/README.md

```diff
@@ -0,0 +1,214 @@
+# Resolver Backend
+
+This readme will describe the backend implementation for resolver.
+
+## Ancestry Array
+
+The ancestry array is an array of entity_ids. This array is included with each event sent by the elastic endpoint
+and defines the ancestors of a particular process. The array is formatted such that [0] of the array contains the direct
+parent of the process. [1] of the array contains the grandparent of the process. For example if Process A spawned process
+B which spawned process C. Process C's array would be [B,C].
+
+The ancestry array is currently limited to 20 values. The exact limit should not be relied on though.
+
+The presence of the ancestry array makes querying ancestors and children for a process more efficient.
+
+## Ancestors
+
+To query for ancestors of a process leveraging the ancestry array, we first retrieve the lifecycle events for the event_id
+passed in. Once we have the origin node we can check to see if the document has the `process.Ext.ancestry` array. If
+it does we can perform a search for the values in the array. This will retrieve all the ancestors for the process of interest
+up to the limit of the ancestry array size. Since the array is capped at 20, if the request is asking for more than 20
+ancestors we will have to examine the most distant ancestor that has been retrieved and use its ancestry array to retrieve
+the next set of results to fulfill the request.
+
+### Pagination
+
+After the backend gathers the results for an ancestry query, it will set a pagination cursor depending on the results from ES.
+
+If the number of ancestors we have gathered is equal to the size in the request we don't know if ES has more results or not. So we will set `nextAncestor` to the entity_id of the most distant ancestor retrieved.
+
+If the request asked for 10 and we only found 8 from ES, we know for sure that there aren't anymore results. In this case we will set `nextAncestor` to `null`.
+
+### Code
+
+The code for handling the ancestor logic is in [here](../utils/ancestry_query_handler.ts)
+
+### Ancestors Multiple Queries Example
+
+![alt text](./resolver_tree_ancestry.png 'Retrieve ancestors')
+
+For this example let's assume that the ancestry array limit is 2. The process of interest is A (the entity_id of a node is the character in the circle). Process A has an ancestry array of `[3,2]`, its parent has an ancestry array of `[2,1]` etc. Here is the execution of a request for 3 ancestors for entity_id A.
```

Comment:
:grey_question:  This concept ('limit of the ancestry array size'/'the ancestry array limit') is referred to in a few different ways. To make it easier on the reader, I'd suggest picking one term putting it in bold/emphasis and using that consistently throughout.

Reply:
Good idea